python/ray/data/_internal/pipelined_dataset_iterator.py

```diff
@@ -0,0 +1,103 @@
+from typing import TYPE_CHECKING, Dict, List, Optional, Union, Iterator
+
+from ray.data import Dataset
+from ray.data.dataset_iterator import DatasetIterator
+from ray.data._internal.block_batching import BatchType
+from ray.data._internal.torch_iterable_dataset import TorchTensorBatchType
+
+if TYPE_CHECKING:
+    import tf
+    import torch
+    from ray.data import DatasetPipeline
+
+
+class PipelinedDatasetIterator(DatasetIterator):
+    def __init__(
+        self,
+        base_dataset_pipeline: "DatasetPipeline",
+    ):
+        self._base_dataset_pipeline = base_dataset_pipeline
+        self._epoch_iterator = None
+
+    def _get_next_dataset(self) -> "DatasetPipeline":
```

Comment:
Would it make sense to name this `_get_next_dataset_pipeline`? The type annotation if for `DatasetPipeline`, but the name made me think it's a `Dataset`.

Reply:
Ah my bad, this should be `Dataset`. Thanks for the catch!